# PROJECT_LESSONS.md

–§–∞–π–ª –¥–ª—è —Ö—Ä–∞–Ω–µ–Ω–∏—è –≤–∞–∂–Ω—ã—Ö —É—Ä–æ–∫–æ–≤, –ø—Ä–∞–≤–∏–ª –∏ —Å—Ç–∞–Ω–¥–∞—Ä—Ç–æ–≤ —Ä–∞–∑—Ä–∞–±–æ—Ç–∫–∏ –ø—Ä–æ–µ–∫—Ç–∞ SurebetCalculator.

---

## üîß –ü—Ä–∞–≤–∏–ª–∞ —Å–±–æ—Ä–∫–∏ –∏ –ø—Ä–æ–≤–µ—Ä–∫–∏

### –ü—Ä–æ–≤–µ—Ä–∫–∞ –∫–æ–º–ø–∏–ª—è—Ü–∏–∏
- **–ü—Ä–∞–≤–∏–ª–æ:** –î–ª—è –ø—Ä–æ–≤–µ—Ä–∫–∏ –∫–æ–º–ø–∏–ª—è—Ü–∏–∏ –∏—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å —Å–∏–º—É–ª—è—Ç–æ—Ä iPhone 17 Pro iOS 26.2 (23C54) —Å –∏–¥–µ–Ω—Ç–∏—Ñ–∏–∫–∞—Ç–æ—Ä–æ–º `F8F50881-5D0E-49DA-AA54-1312A752EED9`
- **–ö–æ–º–∞–Ω–¥–∞:** `xcodebuild -project surebet-calculator.xcodeproj -scheme surebet-calculator -destination 'id=F8F50881-5D0E-49DA-AA54-1312A752EED9' build`
- **–î–ª—è Swift Package Manager:** –ò—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å `xcodebuild -resolvePackageDependencies` –≤–º–µ—Å—Ç–æ `swift build` –¥–ª—è –ø–∞–∫–µ—Ç–æ–≤ —Å `SwiftLintBuildToolPlugin` (–∫—Ä–æ—Å—Å-—Å–±–æ—Ä–∫–∞ –¥–∞—ë—Ç –ª–æ–∂–Ω—ã–µ –ø–∞–¥–µ–Ω–∏—è)

### –ü—Ä–æ–≤–µ—Ä–∫–∞ —Ç–µ—Å—Ç–æ–≤
- **–ö–†–ò–¢–ò–ß–ù–û:** –ü–æ—Å–ª–µ –ª—é–±—ã—Ö –∏–∑–º–µ–Ω–µ–Ω–∏–π –≤ —Ç–µ—Å—Ç–∞—Ö –û–ë–Ø–ó–ê–¢–ï–õ–¨–ù–û –ø—Ä–æ–≤–µ—Ä—è—Ç—å –Ω–µ —Ç–æ–ª—å–∫–æ –∫–æ–º–ø–∏–ª—è—Ü–∏—é, –Ω–æ –∏ –í–´–ü–û–õ–ù–ï–ù–ò–ï —Ç–µ—Å—Ç–æ–≤
- **–ö–æ–º–∞–Ω–¥–∞:** `xcodebuild test -project surebet-calculator.xcodeproj -scheme surebet-calculator -destination 'id=F8F50881-5D0E-49DA-AA54-1312A752EED9'`
- **–ö—Ä–∏—Ç–µ—Ä–∏–π —É—Å–ø–µ—Ö–∞:** `TEST SUCCEEDED` –≤ –≤—ã–≤–æ–¥–µ

### –ü—Ä–æ–≤–µ—Ä–∫–∞ –ª–∏–Ω—Ç–µ—Ä–∞
- **–û–ë–Ø–ó–ê–¢–ï–õ–¨–ù–û:** –ü–µ—Ä–µ–¥ –∫–æ–º–º–∏—Ç–æ–º –ø—Ä–æ–≤–µ—Ä—è—Ç—å –≤–æ—Ä–Ω–∏–Ω–≥–∏ –ª–∏–Ω—Ç–µ—Ä–∞ —á–µ—Ä–µ–∑ `read_lints`
- **–ò—Å–ø—Ä–∞–≤–ª—è—Ç—å:** Trailing whitespace, line length violations, nesting violations
- **SwiftLint –æ—à–∏–±–∫–∏ –≤ —Ç–µ—Å—Ç–∞—Ö:** –ò—Å–ø—Ä–∞–≤–ª—è—Ç—å –∏–ª–∏ –æ—Ç–∫–ª—é—á–∞—Ç—å —á–µ—Ä–µ–∑ `// swiftlint:disable:next` (–Ω–µ blanket disable)

---

## üèóÔ∏è –ê—Ä—Ö–∏—Ç–µ–∫—Ç—É—Ä–Ω—ã–µ –ø—Ä–∞–≤–∏–ª–∞

### –ó–∞–≤–∏—Å–∏–º–æ—Å—Ç–∏ –º–æ–¥—É–ª–µ–π
- **–ü—Ä–∞–≤–∏–ª–æ:** –ü—Ä–∏ —Å–æ–∑–¥–∞–Ω–∏–∏ –∫–æ–Ω—Å—Ç–∞–Ω—Ç —É—á–∏—Ç—ã–≤–∞—Ç—å –∞—Ä—Ö–∏—Ç–µ–∫—Ç—É—Ä—É –º–æ–¥—É–ª–µ–π - –Ω–µ —Å–æ–∑–¥–∞–≤–∞—Ç—å —Ü–∏–∫–ª–∏—á–µ—Å–∫–∏—Ö –∑–∞–≤–∏—Å–∏–º–æ—Å—Ç–µ–π
- **–ü—Ä–∏–º–µ—Ä:** Onboarding –Ω–µ –¥–æ–ª–∂–µ–Ω –∑–∞–≤–∏—Å–µ—Ç—å –æ—Ç SurebetCalculator, –ø–æ—ç—Ç–æ–º—É —Å–æ–∑–¥–∞–Ω –æ—Ç–¥–µ–ª—å–Ω—ã–π `OnboardingConstants.swift`

### –ú–æ–∫–∏ –¥–ª—è —Ç–µ—Å—Ç–æ–≤
- **–ü—Ä–∞–≤–∏–ª–æ:** –ü—Ä–∏ —Å–æ–∑–¥–∞–Ω–∏–∏ –º–æ–∫–æ–≤ –ø—Ä–æ–≤–µ—Ä—è—Ç—å –∑–∞–≤–∏—Å–∏–º–æ—Å—Ç–∏ —Ç–µ—Å—Ç–æ–≤–æ–≥–æ —Ç–∞—Ä–≥–µ—Ç–∞ –≤ Package.swift
- **–ù–µ —Å–æ–∑–¥–∞–≤–∞—Ç—å:** –ú–æ–∫–∏ –¥–ª—è –º–æ–¥—É–ª–µ–π, –∫–æ—Ç–æ—Ä—ã–µ –Ω–µ —è–≤–ª—è—é—Ç—Å—è –∑–∞–≤–∏—Å–∏–º–æ—Å—Ç—è–º–∏ —Ç–µ—Å—Ç–æ–≤–æ–≥–æ —Ç–∞—Ä–≥–µ—Ç–∞
- **SwiftLint:** –ü—Ä–∞–≤–∏–ª–æ `large_tuple` –æ–≥—Ä–∞–Ω–∏—á–∏–≤–∞–µ—Ç –∫–æ—Ä—Ç–µ–∂–∏ –º–∞–∫—Å–∏–º—É–º 2 —ç–ª–µ–º–µ–Ω—Ç–∞–º–∏ - –∏—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å —Å—Ç—Ä—É–∫—Ç—É—Ä—ã –¥–ª—è —Ö—Ä–∞–Ω–µ–Ω–∏—è –±–æ–ª—å—à–µ–≥–æ –∫–æ–ª–∏—á–µ—Å—Ç–≤–∞ –¥–∞–Ω–Ω—ã—Ö

### MainActor –∏ —Ç–µ—Å—Ç—ã
- **–ü—Ä–∞–≤–∏–ª–æ:** –ï—Å–ª–∏ ViewModel –ø–æ–º–µ—á–µ–Ω `@MainActor`, —Ç–µ—Å—Ç–æ–≤—ã–π –∫–ª–∞—Å—Å —Ç–æ–∂–µ –¥–æ–ª–∂–µ–Ω –±—ã—Ç—å –ø–æ–º–µ—á–µ–Ω `@MainActor`
- **–ü—Ä–∏—á–∏–Ω–∞:** –ò–Ω–∞—á–µ –±—É–¥—É—Ç –æ—à–∏–±–∫–∏ –∫–æ–º–ø–∏–ª—è—Ü–∏–∏ Swift 6 –æ MainActor isolation

### UIDevice –≤ Swift 6
- **–ü—Ä–æ–±–ª–µ–º–∞:** `UIDevice.current` —Ç—Ä–µ–±—É–µ—Ç MainActor –≤ Swift 6, –Ω–æ –∏—Å–ø–æ–ª—å–∑—É–µ—Ç—Å—è –≤ nonisolated –∫–æ–Ω—Ç–µ–∫—Å—Ç–µ
- **–†–µ—à–µ–Ω–∏–µ:** –°–æ–∑–¥–∞—Ç—å –¥–≤–∞ —Å–≤–æ–π—Å—Ç–≤–∞ –≤ Device:
  - `@MainActor static var isIPad: Bool` - –¥–ª—è –∏—Å–ø–æ–ª—å–∑–æ–≤–∞–Ω–∏—è –∏–∑ MainActor –∫–æ–Ω—Ç–µ–∫—Å—Ç–∞
  - `nonisolated(unsafe) static var isIPadUnsafe: Bool` - –¥–ª—è –∏—Å–ø–æ–ª—å–∑–æ–≤–∞–Ω–∏—è –∏–∑ nonisolated –∫–æ–Ω—Ç–µ–∫—Å—Ç–∞, –∏—Å–ø–æ–ª—å–∑—É—è `MainActor.assumeIsolated` –¥–ª—è –æ–±—Ö–æ–¥–∞ –ø—Ä–æ–≤–µ—Ä–∫–∏
- **–£—Ä–æ–∫:** –ü—Ä–∏ –∏—Å–ø–æ–ª—å–∑–æ–≤–∞–Ω–∏–∏ UIDevice.current –≤ Swift 6 –∏–∑ nonisolated –∫–æ–Ω—Ç–µ–∫—Å—Ç–∞ –Ω—É–∂–Ω–æ –∏—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å `MainActor.assumeIsolated` –≤–Ω—É—Ç—Ä–∏ `nonisolated(unsafe)` —Å–≤–æ–π—Å—Ç–≤–∞

---

## üìù –°—Ç–∞–Ω–¥–∞—Ä—Ç—ã –∫–æ–¥–∞ –∏ —Å—Ç–∏–ª—å

### Swift 6 Concurrency
- –ò—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å `async/await`, `Task`, `Actor` –≤–º–µ—Å—Ç–æ —Å—Ç–∞—Ä—ã—Ö –º–µ—Ö–∞–Ω–∏–∑–º–æ–≤
- –í—Å–µ ViewModel –¥–æ–ª–∂–Ω—ã –±—ã—Ç—å –ø–æ–º–µ—á–µ–Ω—ã `@MainActor`
- –ü—Ä–æ–≤–µ—Ä—è—Ç—å `Sendable` –∫–æ–Ω—Ñ–æ—Ä–º–Ω–æ—Å—Ç—å –¥–ª—è –≤—Å–µ—Ö –º–æ–¥–µ–ª–µ–π –¥–∞–Ω–Ω—ã—Ö –∏ —Å–µ—Ä–≤–∏—Å–æ–≤ —Å async –º–µ—Ç–æ–¥–∞–º–∏
- **–ü—Ä–∞–≤–∏–ª–æ:** `NSLock` —É–∂–µ —è–≤–ª—è–µ—Ç—Å—è `Sendable`, –Ω–µ –Ω—É–∂–Ω–æ –∏—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å `nonisolated(unsafe)` –¥–ª—è –∫–æ–Ω—Å—Ç–∞–Ω—Ç —Ç–∏–ø–∞ `NSLock`
- **–ü—Ä–∞–≤–∏–ª–æ:** –ù–µ –∏—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å `swiftlint:disable:next sendable_warning` - —ç—Ç–æ –Ω–µ –≤–∞–ª–∏–¥–Ω–æ–µ –ø—Ä–∞–≤–∏–ª–æ SwiftLint

### –¢–∏–ø–∏–∑–∞—Ü–∏—è
- –ò—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å —Å—Ç—Ä–æ–≥—É—é —Ç–∏–ø–∏–∑–∞—Ü–∏—é
- –ò–∑–±–µ–≥–∞—Ç—å `Any` –∏ –Ω–µ–±–µ–∑–æ–ø–∞—Å–Ω—ã—Ö –ø—Ä–∏–≤–µ–¥–µ–Ω–∏–π —Ç–∏–ø–æ–≤ (`as!`)
- –ò—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å `guard let` –∏–ª–∏ `if let` –¥–ª—è –±–µ–∑–æ–ø–∞—Å–Ω—ã—Ö –ø—Ä–∏–≤–µ–¥–µ–Ω–∏–π

### –°—Ç—Ä—É–∫—Ç—É—Ä—ã vs –ö–ª–∞—Å—Å—ã
- –ü–æ —É–º–æ–ª—á–∞–Ω–∏—é –∏—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å `struct`
- –ï—Å–ª–∏ –Ω—É–∂–µ–Ω –∫–ª–∞—Å—Å, –ø–æ–º–µ—á–∞—Ç—å –µ–≥–æ `final`, –µ—Å–ª–∏ –æ–Ω –Ω–µ –ø—Ä–µ–¥–Ω–∞–∑–Ω–∞—á–µ–Ω –¥–ª—è –Ω–∞—Å–ª–µ–¥–æ–≤–∞–Ω–∏—è

### –ú–æ–¥–∏—Ñ–∏–∫–∞—Ç–æ—Ä—ã –¥–æ—Å—Ç—É–ø–∞
- –í—Å–µ–≥–¥–∞ —è–≤–Ω–æ —É–∫–∞–∑—ã–≤–∞—Ç—å –º–æ–¥–∏—Ñ–∏–∫–∞—Ç–æ—Ä—ã –¥–æ—Å—Ç—É–ø–∞ (`private`, `fileprivate`, `public`)
- –ü–æ —É–º–æ–ª—á–∞–Ω–∏—é —Å–≤–æ–π—Å—Ç–≤–∞ –≤–Ω—É—Ç—Ä–∏ View –¥–æ–ª–∂–Ω—ã –±—ã—Ç—å `private`

### SOLID –ø—Ä–∏–Ω—Ü–∏–ø—ã
- **S (Single Responsibility):** –ö–∞–∂–¥—ã–π –∫–æ–º–ø–æ–Ω–µ–Ω—Ç (View, ViewModel, Service) –¥–æ–ª–∂–µ–Ω –æ—Ç–≤–µ—á–∞—Ç—å –∑–∞ –æ–¥–Ω—É –∑–∞–¥–∞—á—É
- **O (Open/Closed):** –ö–æ–¥ –¥–æ–ª–∂–µ–Ω –±—ã—Ç—å –æ—Ç–∫—Ä—ã—Ç –¥–ª—è —Ä–∞—Å—à–∏—Ä–µ–Ω–∏—è, –Ω–æ –∑–∞–∫—Ä—ã—Ç –¥–ª—è –º–æ–¥–∏—Ñ–∏–∫–∞—Ü–∏–∏ (–∏—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å –ø—Ä–æ—Ç–æ–∫–æ–ª—ã)
- **D (Dependency Inversion):** –í–Ω–µ–¥—Ä–µ–Ω–∏–µ –∑–∞–≤–∏—Å–∏–º–æ—Å—Ç–µ–π —á–µ—Ä–µ–∑ init, –∏—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å –ø—Ä–æ—Ç–æ–∫–æ–ª—ã –¥–ª—è –∏–Ω–≤–µ—Ä—Å–∏–∏ –∑–∞–≤–∏—Å–∏–º–æ—Å—Ç–µ–π

### UI –∏ SwiftUI
- –†–∞–∑–¥–µ–ª—è—Ç—å –±–æ–ª—å—à–∏–µ `View` –Ω–∞ –º–µ–ª–∫–∏–µ –∫–æ–º–ø–æ–Ω–µ–Ω—Ç—ã
- –ï—Å–ª–∏ —Ç–µ–ª–æ `body` –ø—Ä–µ–≤—ã—à–∞–µ—Ç 20 —Å—Ç—Ä–æ–∫ - –≤—ã–Ω–æ—Å–∏—Ç—å –∫–æ–¥ –≤ –æ—Ç–¥–µ–ª—å–Ω–æ–µ —Å–≤–æ–π—Å—Ç–≤–æ –∏–ª–∏ —Å—Ç—Ä—É–∫—Ç—É—Ä—É
- –ò—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å `#Preview` –º–∞–∫—Ä–æ—Å –¥–ª—è –≤—Å–µ—Ö UI –∫–æ–º–ø–æ–Ω–µ–Ω—Ç–æ–≤

### –õ–æ–∫–∞–ª–∏–∑–∞—Ü–∏—è
- **–°–¢–†–û–ì–ò–ô –ó–ê–ü–†–ï–¢** –Ω–∞ —Ö–∞—Ä–¥–∫–æ–¥ —Å—Ç—Ä–æ–∫ –≤ UI
- –í–µ—Å—å —Ç–µ–∫—Å—Ç –¥–æ–ª–∂–µ–Ω –±—ã—Ç—å –≤—ã–Ω–µ—Å–µ–Ω –≤ —Ñ–∞–π–ª—ã –ª–æ–∫–∞–ª–∏–∑–∞—Ü–∏–∏ (String Catalogs `.xcstrings`)
- –ò—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å `String(localized:)` –¥–ª—è –≤—Å–µ—Ö —Å—Ç—Ä–æ–∫ –≤ UI

### –ö–æ–Ω—Å—Ç–∞–Ω—Ç—ã
- –í—ã–Ω–æ—Å–∏—Ç—å –≤—Å–µ –º–∞–≥–∏—á–µ—Å–∫–∏–µ —á–∏—Å–ª–∞, —Ü–≤–µ—Ç–∞, —à—Ä–∏—Ñ—Ç—ã –∏ —Ä–∞–∑–º–µ—Ä—ã –≤ —Å—Ç–∞—Ç–∏—á–µ—Å–∫–∏–µ –∫–æ–Ω—Å—Ç–∞–Ω—Ç—ã
- –ò—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å `enum` –∏–ª–∏ `struct` –¥–ª—è –≥—Ä—É–ø–ø–∏—Ä–æ–≤–∫–∏ –∫–æ–Ω—Å—Ç–∞–Ω—Ç –ø–æ –∫–∞—Ç–µ–≥–æ—Ä–∏—è–º
- –ü—Ä–∏–º–µ—Ä: `static let defaultPadding: CGFloat = 16.0`

### –î–æ–∫—É–º–µ–Ω—Ç–∞—Ü–∏—è
- –í—Å—è –¥–æ–∫—É–º–µ–Ω—Ç–∞—Ü–∏—è –∏ –∫–æ–º–º–µ–Ω—Ç–∞—Ä–∏–∏ –¥–æ–ª–∂–Ω—ã –±—ã—Ç—å –Ω–∞ **–†–£–°–°–ö–û–ú –Ø–ó–´–ö–ï**
- –ò—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å —Ñ–æ—Ä–º–∞—Ç Swift Doc (`///`) –¥–ª—è –ø—É–±–ª–∏—á–Ω—ã—Ö –º–µ—Ç–æ–¥–æ–≤, –∫–ª–∞—Å—Å–æ–≤ –∏ —Å—Ç—Ä—É–∫—Ç—É—Ä
- –û–ø–∏—Å—ã–≤–∞—Ç—å *–ø–æ—á–µ–º—É* —Å–¥–µ–ª–∞–Ω–æ —Ç–∞–∫, –∞ –Ω–µ *—á—Ç–æ* –¥–µ–ª–∞–µ—Ç –∫–æ–¥
- –ö–æ–º–º–µ–Ω—Ç–∏—Ä–æ–≤–∞—Ç—å —Å–ª–æ–∂–Ω—ã–µ –∞–ª–≥–æ—Ä–∏—Ç–º—ã, workarounds –∏ –Ω–µ–æ—á–µ–≤–∏–¥–Ω—É—é –ª–æ–≥–∏–∫—É

### –°—Ç—Ä—É–∫—Ç—É—Ä–∞ —Ñ–∞–π–ª–æ–≤
- –ò—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å `MARK: - Section Name` –¥–ª—è —Ä–∞–∑–¥–µ–ª–µ–Ω–∏—è —Å–≤–æ–π—Å—Ç–≤, –º–µ—Ç–æ–¥–æ–≤ –∂–∏–∑–Ω–µ–Ω–Ω–æ–≥–æ —Ü–∏–∫–ª–∞, –ø—É–±–ª–∏—á–Ω—ã—Ö –∏ –ø—Ä–∏–≤–∞—Ç–Ω—ã—Ö –º–µ—Ç–æ–¥–æ–≤
- –ü–æ—Ä—è–¥–æ–∫ –≤ —Ñ–∞–π–ª–µ:
  1. –°–≤–æ–π—Å—Ç–≤–∞ (`@Environment`, `@State`, `@Binding`, –æ–±—ã—á–Ω—ã–µ let/var)
  2. `init` (–µ—Å–ª–∏ –µ—Å—Ç—å)
  3. `body` (–¥–ª—è SwiftUI)
  4. –ü—É–±–ª–∏—á–Ω—ã–µ –º–µ—Ç–æ–¥—ã
  5. –ü—Ä–∏–≤–∞—Ç–Ω—ã–µ –º–µ—Ç–æ–¥—ã

---

## üöÄ –ü—Ä–∞–≤–∏–ª–∞ –∫–∞—á–µ—Å—Ç–≤–µ–Ω–Ω–æ–π —Ä–∞–∑—Ä–∞–±–æ—Ç–∫–∏

### –ü–µ—Ä–µ–¥ –∫–æ–º–º–∏—Ç–æ–º
1. **–û–ë–Ø–ó–ê–¢–ï–õ–¨–ù–û:** –ü—Ä–æ–≤–µ—Ä–∏—Ç—å —Å–±–æ—Ä–∫—É –ø—Ä–æ–µ–∫—Ç–∞ —á–µ—Ä–µ–∑ `xcodebuild`
2. **–û–ë–Ø–ó–ê–¢–ï–õ–¨–ù–û:** –ü—Ä–æ–≤–µ—Ä–∏—Ç—å –≤–æ—Ä–Ω–∏–Ω–≥–∏ –ª–∏–Ω—Ç–µ—Ä–∞ —á–µ—Ä–µ–∑ `read_lints`
3. **–û–ë–Ø–ó–ê–¢–ï–õ–¨–ù–û:** –ò—Å–ø—Ä–∞–≤–∏—Ç—å –≤—Å–µ –æ—à–∏–±–∫–∏ –∏ –≤–æ—Ä–Ω–∏–Ω–≥–∏ –ø–µ—Ä–µ–¥ –∫–æ–º–º–∏—Ç–æ–º
4. **–û–ë–Ø–ó–ê–¢–ï–õ–¨–ù–û:** –ï—Å–ª–∏ –∏–∑–º–µ–Ω—è–ª–∏—Å—å —Ç–µ—Å—Ç—ã - –ø—Ä–æ–≤–µ—Ä–∏—Ç—å –∏—Ö –≤—ã–ø–æ–ª–Ω–µ–Ω–∏–µ —á–µ—Ä–µ–∑ `xcodebuild test`

### –ü—Ä–∏ —Å–æ–∑–¥–∞–Ω–∏–∏ –Ω–æ–≤—ã—Ö —Ñ–∞–π–ª–æ–≤
- –ü—Ä–æ–≤–µ—Ä—è—Ç—å trailing whitespace
- –ü—Ä–æ–≤–µ—Ä—è—Ç—å –¥–ª–∏–Ω—É —Å—Ç—Ä–æ–∫ (line_length violation)
- –î–æ–±–∞–≤–ª—è—Ç—å MARK —Ä–∞–∑–¥–µ–ª—ã
- –î–æ–±–∞–≤–ª—è—Ç—å –º–æ–¥–∏—Ñ–∏–∫–∞—Ç–æ—Ä—ã –¥–æ—Å—Ç—É–ø–∞
- –î–æ–±–∞–≤–ª—è—Ç—å –¥–æ–∫—É–º–µ–Ω—Ç–∞—Ü–∏—é –Ω–∞ —Ä—É—Å—Å–∫–æ–º —è–∑—ã–∫–µ

### –ü—Ä–∏ —Ä–∞–±–æ—Ç–µ —Å –∑–∞–≤–∏—Å–∏–º–æ—Å—Ç—è–º–∏
- –ü—Ä–æ–≤–µ—Ä—è—Ç—å –∑–∞–≤–∏—Å–∏–º–æ—Å—Ç–∏ —Ç–µ—Å—Ç–æ–≤–æ–≥–æ —Ç–∞—Ä–≥–µ—Ç–∞ –ø–µ—Ä–µ–¥ —Å–æ–∑–¥–∞–Ω–∏–µ–º –º–æ–∫–æ–≤
- –ò–∑–±–µ–≥–∞—Ç—å —Ü–∏–∫–ª–∏—á–µ—Å–∫–∏—Ö –∑–∞–≤–∏—Å–∏–º–æ—Å—Ç–µ–π –º–µ–∂–¥—É –º–æ–¥—É–ª—è–º–∏
- –ò—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å Dependency Injection —á–µ—Ä–µ–∑ init

### –ü—Ä–∏ —Ä–∞–±–æ—Ç–µ —Å —Ç–µ—Å—Ç–∞–º–∏
- –¢–µ—Å—Ç–æ–≤—ã–π –∫–ª–∞—Å—Å –¥–æ–ª–∂–µ–Ω –±—ã—Ç—å –ø–æ–º–µ—á–µ–Ω `@MainActor`, –µ—Å–ª–∏ —Ç–µ—Å—Ç–∏—Ä—É–µ–º—ã–π –∫–ª–∞—Å—Å –ø–æ–º–µ—á–µ–Ω `@MainActor`
- –ò—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å —Å—Ç—Ä—É–∫—Ç—É—Ä—ã –≤–º–µ—Å—Ç–æ –∫–æ—Ä—Ç–µ–∂–µ–π —Å –±–æ–ª–µ–µ —á–µ–º 2 —ç–ª–µ–º–µ–Ω—Ç–∞–º–∏
- –ü—Ä–æ–≤–µ—Ä—è—Ç—å –Ω–µ —Ç–æ–ª—å–∫–æ –∫–æ–º–ø–∏–ª—è—Ü–∏—é, –Ω–æ –∏ –≤—ã–ø–æ–ª–Ω–µ–Ω–∏–µ —Ç–µ—Å—Ç–æ–≤
- **–ö–†–ò–¢–ò–ß–ù–û:** –ü—Ä–∏ –∏—Å–ø–æ–ª—å–∑–æ–≤–∞–Ω–∏–∏ MockURLProtocol –≤ –ø–∞—Ä–∞–ª–ª–µ–ª—å–Ω—ã—Ö —Ç–µ—Å—Ç–∞—Ö –∏—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å –ø–æ—Ç–æ–∫–æ–±–µ–∑–æ–ø–∞—Å–Ω—É—é –º–∞—Ä—à—Ä—É—Ç–∏–∑–∞—Ü–∏—é –ø–æ URL, –∞ –Ω–µ –æ–¥–∏–Ω –≥–ª–æ–±–∞–ª—å–Ω—ã–π —Ö–µ–Ω–¥–ª–µ—Ä
- **–ü—Ä–∞–≤–∏–ª–æ:** –†–µ–≥–∏—Å—Ç—Ä–∏—Ä–æ–≤–∞—Ç—å —Ö–µ–Ω–¥–ª–µ—Ä—ã –¥–ª—è –ø–æ–ª–Ω–æ–≥–æ URL (—Å –ø—É—Ç–µ–º), –∞ –Ω–µ —Ç–æ–ª—å–∫–æ baseURL, —Ç–∞–∫ –∫–∞–∫ Service –¥–µ–ª–∞–µ—Ç –∑–∞–ø—Ä–æ—Å—ã –∫ `baseURL.appendingPathComponent("banner")`
- **–ü—Ä–∞–≤–∏–ª–æ:** –£–¥–∞–ª—è—Ç—å –Ω–µ–∏—Å–ø–æ–ª—å–∑—É–µ–º—ã–µ –ø–µ—Ä–µ–º–µ–Ω–Ω—ã–µ –≤ —Ç–µ—Å—Ç–∞—Ö (–Ω–∞–ø—Ä–∏–º–µ—Ä, `bannerData`, –µ—Å–ª–∏ –∏—Å–ø–æ–ª—å–∑—É–µ—Ç—Å—è `bannerDataWithUniqueImage`)
- **–ü—Ä–∞–≤–∏–ª–æ:** –í unit/integration —Ç–µ—Å—Ç–∞—Ö –∑–∞–ø—Ä–µ—â–µ–Ω–æ –∏—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å `Thread.sleep`/`Task.sleep`. –ò—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å –æ–∂–∏–¥–∞–Ω–∏–µ —Å–æ—Å—Ç–æ—è–Ω–∏—è/–≤—ã–∑–æ–≤–æ–≤ –∏–ª–∏ DI –≤—Ä–µ–º–µ–Ω–∏ (Delay/Clock) —Å immediate/fake —Ä–µ–∞–ª–∏–∑–∞—Ü–∏–µ–π.
- **–ü—Ä–∞–≤–∏–ª–æ:** –í UI —Ç–µ—Å—Ç–∞—Ö –∑–∞–ø—Ä–µ—â–µ–Ω–æ –∏—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å `Thread.sleep`. –ò—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å `waitForExistence(timeout:)` –∏–ª–∏ –æ–∂–∏–¥–∞–Ω–∏—è —á–µ—Ä–µ–∑ `XCTNSPredicateExpectation`/`XCTWaiter`.

---

## üìö –ò—Å—Ç–æ—Ä–∏—è –≤–∞–∂–Ω—ã—Ö –æ—à–∏–±–æ–∫ –∏ —Ä–µ—à–µ–Ω–∏–π

### 2026-02-01: SwiftUI transition —Ç–µ—Ä—è–µ—Ç—Å—è –∏–ª–∏ –ø–æ–ª—É—á–∞–µ—Ç opacity –ø—Ä–∏ –∑–∞–∫—Ä—ã—Ç–∏–∏ –æ–Ω–±–æ—Ä–¥–∏–Ω–≥–∞
- **–ü—Ä–æ–±–ª–µ–º–∞:** –ü—Ä–∏ –∑–∞–∫—Ä—ã—Ç–∏–∏ –æ–Ω–±–æ—Ä–¥–∏–Ω–≥–∞ `.move(edge: .bottom)` —Ä–∞–±–æ—Ç–∞–ª –±–µ–∑ –∞–Ω–∏–º–∞—Ü–∏–∏ –∏–ª–∏ –¥–æ–±–∞–≤–ª—è–ª–∞—Å—å –Ω–µ–∂–µ–ª–∞—Ç–µ–ª—å–Ω–∞—è opacity
- **–ü—Ä–∏—á–∏–Ω–∞:** –ì–ª–æ–±–∞–ª—å–Ω–∞—è –∞–Ω–∏–º–∞—Ü–∏—è –Ω–∞ –∫–æ–Ω—Ç–µ–π–Ω–µ—Ä–µ/—Å–º–µ–Ω–∞ –≤–µ—Ç–æ–∫ `if` –ø—Ä–æ–≤–æ—Ü–∏—Ä–æ–≤–∞–ª–∞ —Å–∏—Å—Ç–µ–º–Ω—ã–π crossfade; transition –ø—Ä–∏–º–µ–Ω—è–ª—Å—è –Ω–µ –∫ —Ç–æ–º—É —É—Ä–æ–≤–Ω—é
- **–†–µ—à–µ–Ω–∏–µ:** –î–µ—Ä–∂–∞—Ç—å –±–∞–∑–æ–≤—ã–π —ç–∫—Ä–∞–Ω –ø–æ—Å—Ç–æ—è–Ω–Ω–æ (–Ω–∞–ø—Ä–∏–º–µ—Ä, —á–µ—Ä–µ–∑ `ZStack`), –∞ transition –∏ –∞–Ω–∏–º–∞—Ü–∏—é –≤–µ—à–∞—Ç—å —Ç–æ–ª—å–∫–æ –Ω–∞ —Å–ª–æ–π –æ–Ω–±–æ—Ä–¥–∏–Ω–≥–∞
- **–£—Ä–æ–∫:** –î–ª—è –∫–æ—Ä—Ä–µ–∫—Ç–Ω–æ–≥–æ move‚Äëtransition –±–µ–∑ opacity –Ω–µ–ª—å–∑—è –∞–Ω–∏–º–∏—Ä–æ–≤–∞—Ç—å –≤–µ—Å—å –∫–æ–Ω—Ç–µ–π–Ω–µ—Ä ‚Äî –∞–Ω–∏–º–∏—Ä—É–π—Ç–µ —Ç–æ–ª—å–∫–æ —É–¥–∞–ª—è–µ–º—ã–π/–≤—Å—Ç–∞–≤–ª—è–µ–º—ã–π view

### 2026-01-31: –î–µ—Ç–µ—Ä–º–∏–Ω–∏—Ä–æ–≤–∞–Ω–Ω—ã–µ –æ–∂–∏–¥–∞–Ω–∏—è –≤–º–µ—Å—Ç–æ sleep
- **–ü—Ä–æ–±–ª–µ–º–∞:** `Thread.sleep`/`Task.sleep` –≤ —Ç–µ—Å—Ç–∞—Ö –ø—Ä–∏–≤–æ–¥–∏–ª–∏ –∫ —Ñ–ª–µ–π–∫–∞–º –∏ –∑–∞–º–µ–¥–ª—è–ª–∏ CI
- **–†–µ—à–µ–Ω–∏–µ:** –£–±—Ä–∞—Ç—å sleep –∏–∑ unit/integration/UI —Ç–µ—Å—Ç–æ–≤, –ø–µ—Ä–µ–π—Ç–∏ –Ω–∞ –æ–∂–∏–¥–∞–Ω–∏–µ —Å–æ—Å—Ç–æ—è–Ω–∏—è/–≤—ã–∑–æ–≤–æ–≤ –∏ DI –≤—Ä–µ–º–µ–Ω–∏ (Delay/Clock) —Å immediate/fake —Ä–µ–∞–ª–∏–∑–∞—Ü–∏–µ–π
- **–£—Ä–æ–∫:** –ú–æ–∫–∏ –Ω–µ –¥–æ–ª–∂–Ω—ã —Å–∏–º—É–ª–∏—Ä–æ–≤–∞—Ç—å –∑–∞–¥–µ—Ä–∂–∫–∏; —Å–∏–Ω—Ö—Ä–æ–Ω–∏–∑–∞—Ü–∏—è UI —Ç–µ—Å—Ç–æ–≤ ‚Äî —Ç–æ–ª—å–∫–æ –ø–æ —Å–æ—Å—Ç–æ—è–Ω–∏—é UI
- **–£—Ä–æ–∫:** View –Ω–µ –¥–æ–ª–∂–µ–Ω –±—ã—Ç—å DI-–∫–æ–Ω—Ç–µ–π–Ω–µ—Ä–æ–º; —Å–±–æ—Ä–∫–∞ ViewModel ‚Äî –≤ entry-point/factory
- **–£—Ä–æ–∫:** –î–ª—è presentation state –¥–æ–ø—É—Å—Ç–∏–º –∞–¥–∞–ø—Ç–µ—Ä —É—Ä–æ–≤–Ω—è entry-point (–Ω–∞–ø—Ä–∏–º–µ—Ä, `PresentationBinding`), —á—Ç–æ–±—ã ViewModel –Ω–µ –∑–∞–≤–∏—Å–µ–ª–∞ –æ—Ç SwiftUI

### 2026-01-23: Swift Package Manager –∏ SwiftLint
- **–û—à–∏–±–∫–∞:** `swift build` –∫–∞–∫ "–ø—Ä–æ–≤–µ—Ä–∫–∞ —Å–æ–≤–º–µ—Å—Ç–∏–º–æ—Å—Ç–∏" –¥–ª—è iOS-–ø–∞–∫–µ—Ç–∞ —Å `SwiftLintBuildToolPlugin` –¥–∞—ë—Ç –ª–æ–∂–Ω—ã–µ –ø–∞–¥–µ–Ω–∏—è
- **–†–µ—à–µ–Ω–∏–µ:** –ò—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å `xcodebuild -resolvePackageDependencies` –≤–º–µ—Å—Ç–æ `swift build`
- **–£—Ä–æ–∫:** –î–ª—è –ø–∞–∫–µ—Ç–æ–≤ —Å SwiftLint "–≤–∞–ª–∏–¥–Ω–æ—Å—Ç—å" –ø—Ä–æ–≤–µ—Ä—è–µ–º —á–µ—Ä–µ–∑ Xcode-resolve/—Å–±–æ—Ä–∫—É —Å—Ö–µ–º—ã

### 2026-01-24: –ö–æ–Ω—Å—Ç–∞–Ω—Ç—ã –∏ –∑–∞–≤–∏—Å–∏–º–æ—Å—Ç–∏
- **–û—à–∏–±–∫–∞:** –ü–æ—Å–ª–µ —Å–æ–∑–¥–∞–Ω–∏—è —Ñ–∞–π–ª–æ–≤ –∫–æ–Ω—Å—Ç–∞–Ω—Ç –ø—Ä–æ–µ–∫—Ç –Ω–µ —Å–æ–±–∏—Ä–∞–ª—Å—è –∏–∑-–∑–∞ trailing whitespace, line_length violation, —Ü–∏–∫–ª–∏—á–µ—Å–∫–∏—Ö –∑–∞–≤–∏—Å–∏–º–æ—Å—Ç–µ–π
- **–†–µ—à–µ–Ω–∏–µ:** –£–¥–∞–ª–∏—Ç—å trailing whitespace, —Ä–∞–∑–±–∏—Ç—å –¥–ª–∏–Ω–Ω—ã–µ —Å—Ç—Ä–æ–∫–∏, —Å–æ–∑–¥–∞—Ç—å –æ—Ç–¥–µ–ª—å–Ω—ã–π `OnboardingConstants.swift`
- **–£—Ä–æ–∫:** –í—Å–µ–≥–¥–∞ –ø—Ä–æ–≤–µ—Ä—è—Ç—å —Å–±–æ—Ä–∫—É –ø—Ä–æ–µ–∫—Ç–∞ –∏ –≤–æ—Ä–Ω–∏–Ω–≥–∏ –ª–∏–Ω—Ç–µ—Ä–∞ –ø–µ—Ä–µ–¥ –∫–æ–º–º–∏—Ç–æ–º, —É—á–∏—Ç—ã–≤–∞—Ç—å –∞—Ä—Ö–∏—Ç–µ–∫—Ç—É—Ä—É –º–æ–¥—É–ª–µ–π

### 2026-01-24: –ú–æ–∫–∏ –∏ —Ç–µ—Å—Ç—ã
- **–û—à–∏–±–∫–∞:** –ú–æ–∫–∏ –ø—ã—Ç–∞–ª–∏—Å—å –∏–º–ø–æ—Ä—Ç–∏—Ä–æ–≤–∞—Ç—å –º–æ–¥—É–ª–∏, –∫–æ—Ç–æ—Ä—ã–µ –Ω–µ —è–≤–ª—è—é—Ç—Å—è –∑–∞–≤–∏—Å–∏–º–æ—Å—Ç—è–º–∏ —Ç–µ—Å—Ç–æ–≤–æ–≥–æ —Ç–∞—Ä–≥–µ—Ç–∞; Large Tuple Violation; –æ—à–∏–±–∫–∏ MainActor isolation
- **–†–µ—à–µ–Ω–∏–µ:** –£–¥–∞–ª–∏—Ç—å –Ω–µ–∏—Å–ø–æ–ª—å–∑—É–µ–º—ã–µ –º–æ–∫–∏, –∑–∞–º–µ–Ω–∏—Ç—å –∫–æ—Ä—Ç–µ–∂ –Ω–∞ —Å—Ç—Ä—É–∫—Ç—É—Ä—É, –¥–æ–±–∞–≤–∏—Ç—å `@MainActor` –∫ —Ç–µ—Å—Ç–æ–≤–æ–º—É –∫–ª–∞—Å—Å—É
- **–£—Ä–æ–∫:** –ü—Ä–æ–≤–µ—Ä—è—Ç—å –∑–∞–≤–∏—Å–∏–º–æ—Å—Ç–∏ —Ç–µ—Å—Ç–æ–≤–æ–≥–æ —Ç–∞—Ä–≥–µ—Ç–∞, –∏—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å —Å—Ç—Ä—É–∫—Ç—É—Ä—ã –≤–º–µ—Å—Ç–æ –±–æ–ª—å—à–∏—Ö –∫–æ—Ä—Ç–µ–∂–µ–π, —Å–∏–Ω—Ö—Ä–æ–Ω–∏–∑–∏—Ä–æ–≤–∞—Ç—å MainActor –∞–Ω–Ω–æ—Ç–∞—Ü–∏–∏

### 2026-01-27: Race Condition –≤ MockURLProtocol –ø—Ä–∏ –ø–∞—Ä–∞–ª–ª–µ–ª—å–Ω—ã—Ö —Ç–µ—Å—Ç–∞—Ö
- **–û—à–∏–±–∫–∞:** –¢–µ—Å—Ç—ã –ø–∞–¥–∞–ª–∏ —Å –æ—à–∏–±–∫–∞–º–∏ `-1009` –∏ `-1011` —Å–ª—É—á–∞–π–Ω—ã–º –æ–±—Ä–∞–∑–æ–º –ø—Ä–∏ –ø–∞—Ä–∞–ª–ª–µ–ª—å–Ω–æ–º –≤—ã–ø–æ–ª–Ω–µ–Ω–∏–∏
- **–ü—Ä–∏—á–∏–Ω–∞:** –ò—Å–ø–æ–ª—å–∑–æ–≤–∞–ª—Å—è –æ–¥–∏–Ω –≥–ª–æ–±–∞–ª—å–Ω—ã–π `static var requestHandler` –≤ `MockURLProtocol`, –∫–æ—Ç–æ—Ä—ã–π –ø–µ—Ä–µ–∑–∞–ø–∏—Å—ã–≤–∞–ª—Å—è –ø–∞—Ä–∞–ª–ª–µ–ª—å–Ω—ã–º–∏ —Ç–µ—Å—Ç–∞–º–∏ –≤–æ –≤—Ä–µ–º—è –≤—ã–ø–æ–ª–Ω–µ–Ω–∏—è –∑–∞–ø—Ä–æ—Å–∞
- **–†–µ—à–µ–Ω–∏–µ:** –†–µ–∞–ª–∏–∑–æ–≤–∞—Ç—å –ø–æ—Ç–æ–∫–æ–±–µ–∑–æ–ø–∞—Å–Ω—ã–π —Ä–µ–µ—Å—Ç—Ä —Ö–µ–Ω–¥–ª–µ—Ä–æ–≤ –ø–æ URL:
  - –ò—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å `[String: RequestHandler]` —Å–ª–æ–≤–∞—Ä—å —Å –∫–ª—é—á–æ–º `URL.absoluteString`
  - –°–∏–Ω—Ö—Ä–æ–Ω–∏–∑–∏—Ä–æ–≤–∞—Ç—å –¥–æ—Å—Ç—É–ø —á–µ—Ä–µ–∑ `NSLock`
  - –†–µ–≥–∏—Å—Ç—Ä–∏—Ä–æ–≤–∞—Ç—å —Ö–µ–Ω–¥–ª–µ—Ä—ã —á–µ—Ä–µ–∑ `register(url:handler:)` –∏ —É–¥–∞–ª—è—Ç—å —á–µ—Ä–µ–∑ `unregister(url:)`
  - –í `startLoading()` –∏—Å–∫–∞—Ç—å —Ö–µ–Ω–¥–ª–µ—Ä –ø–æ URL –∑–∞–ø—Ä–æ—Å–∞
- **–ö—Ä–∏—Ç–∏—á–Ω–æ:** –†–µ–≥–∏—Å—Ç—Ä–∏—Ä–æ–≤–∞—Ç—å —Ö–µ–Ω–¥–ª–µ—Ä—ã –¥–ª—è –ø–æ–ª–Ω–æ–≥–æ URL (—Å –ø—É—Ç–µ–º), –∞ –Ω–µ —Ç–æ–ª—å–∫–æ baseURL
  - Service –¥–µ–ª–∞–µ—Ç –∑–∞–ø—Ä–æ—Å—ã –∫ `baseURL.appendingPathComponent("banner")`
  - –ù—É–∂–Ω–æ —Ä–µ–≥–∏—Å—Ç—Ä–∏—Ä–æ–≤–∞—Ç—å —Ö–µ–Ω–¥–ª–µ—Ä –¥–ª—è `uniqueBaseURL.appendingPathComponent("banner")`, –∞ –Ω–µ –¥–ª—è `uniqueBaseURL`
- **–£—Ä–æ–∫:** –í Swift 6 —Ç–µ—Å—Ç—ã –≤—ã–ø–æ–ª–Ω—è—é—Ç—Å—è –ø–∞—Ä–∞–ª–ª–µ–ª—å–Ω–æ –ø–æ —É–º–æ–ª—á–∞–Ω–∏—é - –Ω—É–∂–Ω–æ –æ–±–µ—Å–ø–µ—á–∏–≤–∞—Ç—å –ø–æ–ª–Ω—É—é –∏–∑–æ–ª—è—Ü–∏—é —Ç–µ—Å—Ç–æ–≤ —á–µ—Ä–µ–∑ —É–Ω–∏–∫–∞–ª—å–Ω—ã–µ URL –∏ –ø–æ—Ç–æ–∫–æ–±–µ–∑–æ–ø–∞—Å–Ω—ã–µ —Ä–µ–µ—Å—Ç—Ä—ã —Ö–µ–Ω–¥–ª–µ—Ä–æ–≤

### 2026-01-27: SwiftLint –∏ Sendable –ø—Ä–µ–¥—É–ø—Ä–µ–∂–¥–µ–Ω–∏—è
- **–û—à–∏–±–∫–∞:** –ü–æ–ø—ã—Ç–∫–∞ –æ—Ç–∫–ª—é—á–∏—Ç—å –ø—Ä–µ–¥—É–ø—Ä–µ–∂–¥–µ–Ω–∏—è Sendable —á–µ—Ä–µ–∑ `swiftlint:disable:next sendable_warning`
- **–ü—Ä–∏—á–∏–Ω–∞:** `sendable_warning` –Ω–µ —è–≤–ª—è–µ—Ç—Å—è –≤–∞–ª–∏–¥–Ω—ã–º –ø—Ä–∞–≤–∏–ª–æ–º SwiftLint - —ç—Ç–æ –ø—Ä–µ–¥—É–ø—Ä–µ–∂–¥–µ–Ω–∏–µ –∫–æ–º–ø–∏–ª—è—Ç–æ—Ä–∞ Swift 6
- **–†–µ—à–µ–Ω–∏–µ:** –ù–µ –ø—ã—Ç–∞—Ç—å—Å—è –æ—Ç–∫–ª—é—á–∞—Ç—å –ø—Ä–µ–¥—É–ø—Ä–µ–∂–¥–µ–Ω–∏—è Sendable —á–µ—Ä–µ–∑ SwiftLint, –æ–Ω–∏ –Ω–µ –∫—Ä–∏—Ç–∏—á–Ω—ã –¥–ª—è —Ç–µ—Å—Ç–æ–≤–æ–≥–æ –∫–æ–¥–∞
- **–û—à–∏–±–∫–∞:** –ò—Å–ø–æ–ª—å–∑–æ–≤–∞–Ω–∏–µ `nonisolated(unsafe)` –¥–ª—è `NSLock`
- **–ü—Ä–∏—á–∏–Ω–∞:** `NSLock` —É–∂–µ —è–≤–ª—è–µ—Ç—Å—è `Sendable` —Ç–∏–ø–æ–º
- **–†–µ—à–µ–Ω–∏–µ:** –£–±—Ä–∞—Ç—å `nonisolated(unsafe)` –¥–ª—è –∫–æ–Ω—Å—Ç–∞–Ω—Ç —Ç–∏–ø–∞ `NSLock`
- **–£—Ä–æ–∫:** –ü—Ä–æ–≤–µ—Ä—è—Ç—å, —è–≤–ª—è–µ—Ç—Å—è –ª–∏ —Ç–∏–ø —É–∂–µ `Sendable`, –ø–µ—Ä–µ–¥ –∏—Å–ø–æ–ª—å–∑–æ–≤–∞–Ω–∏–µ–º `nonisolated(unsafe)`

### 2026-01-24: UI —Ç–µ—Å—Ç—ã –∏ SwiftUI accessibility identifiers
- **–û—à–∏–±–∫–∞:** `accessibilityIdentifier` –Ω–∞ SwiftUI Button —Å custom label –Ω–µ –≤—Å–µ–≥–¥–∞ –≤–∏–¥–µ–Ω –≤ XCUITest
- **–ü—Ä–∏—á–∏–Ω–∞:** SwiftUI –º–æ–∂–µ—Ç —Å–æ–∑–¥–∞–≤–∞—Ç—å —Å–ª–æ–∂–Ω—É—é –∏–µ—Ä–∞—Ä—Ö–∏—é accessibility, –≥–¥–µ identifier –Ω–µ –ø—Ä–æ–±—Ä–∞—Å—ã–≤–∞–µ—Ç—Å—è
- **–†–µ—à–µ–Ω–∏–µ:** –ò—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å –ø–æ–∏—Å–∫ –ø–æ —Ç–µ–∫—Å—Ç—É –∫–Ω–æ–ø–∫–∏ (`app.buttons["Button Text"]`) –∏–ª–∏ `app.images["system.image.name"]` –¥–ª—è —Å–∏—Å—Ç–µ–º–Ω—ã—Ö –∏–∫–æ–Ω–æ–∫
- **–£—Ä–æ–∫:** –ü—Ä–∏ –Ω–∞–ø–∏—Å–∞–Ω–∏–∏ UI —Ç–µ—Å—Ç–æ–≤ –¥–ª—è SwiftUI:
  - `app.textFields["identifier"]` —Ä–∞–±–æ—Ç–∞–µ—Ç —Ö–æ—Ä–æ—à–æ
  - `app.staticTexts["identifier"]` —Ä–∞–±–æ—Ç–∞–µ—Ç —Ö–æ—Ä–æ—à–æ
  - `app.buttons["Button Text"]` –Ω–∞–¥—ë–∂–Ω–µ–µ —á–µ–º `app.buttons["identifier"]`
  - `app.images["system.name"]` –¥–ª—è —Å–∏—Å—Ç–µ–º–Ω—ã—Ö SF Symbols
  - `app.otherElements["identifier"]` –¥–ª—è –∫–æ–Ω—Ç–µ–π–Ω–µ—Ä–æ–≤ (VStack –∏ —Ç.–¥.)
- **–ö—Ä–∏—Ç–∏—á–Ω–æ:** Launch arguments –¥–ª—è UI —Ç–µ—Å—Ç–æ–≤ –æ–±—Ä–∞–±–∞—Ç—ã–≤–∞—é—Ç—Å—è –≤ AppDelegate —á–µ—Ä–µ–∑ `ProcessInfo.processInfo.arguments`

### 2026-01-24: UI —Ç–µ—Å—Ç—ã –≤ Swift 6 - MainActor isolation
- **–û—à–∏–±–∫–∞:** `Call to main actor-isolated initializer 'init()' in a synchronous nonisolated context`
- **–ü—Ä–∏—á–∏–Ω–∞:** XCUIApplication –∏ –≤—Å–µ –µ–≥–æ –º–µ—Ç–æ–¥—ã –≤ Swift 6 —Ç—Ä–µ–±—É—é—Ç MainActor isolation
- **–†–µ—à–µ–Ω–∏–µ:** –ü–æ–º–µ—Ç–∏—Ç—å —Ç–µ—Å—Ç–æ–≤—ã–π –∫–ª–∞—Å—Å `@MainActor`:
  ```swift
  @MainActor
  final class SurebetCalculatorUITests: XCTestCase { ... }
  ```
- **–£—Ä–æ–∫:** –í Swift 6 –≤—Å–µ UI —Ç–µ—Å—Ç–æ–≤—ã–µ –∫–ª–∞—Å—Å—ã –¥–æ–ª–∂–Ω—ã –±—ã—Ç—å –ø–æ–º–µ—á–µ–Ω—ã `@MainActor`

### 2026-01-24: UI —Ç–µ—Å—Ç—ã - Launch Arguments —Ñ–æ—Ä–º–∞—Ç
- **–û—à–∏–±–∫–∞:** Launch arguments `-resetUserDefaults`, `YES` –Ω–µ —Ä–∞–±–æ—Ç–∞—é—Ç
- **–ü—Ä–∏—á–∏–Ω–∞:** –ù–µ–ø—Ä–∞–≤–∏–ª—å–Ω—ã–π —Ñ–æ—Ä–º–∞—Ç - `YES` –ø–µ—Ä–µ–¥–∞—ë—Ç—Å—è –∫–∞–∫ –æ—Ç–¥–µ–ª—å–Ω—ã–π –∞—Ä–≥—É–º–µ–Ω—Ç
- **–†–µ—à–µ–Ω–∏–µ:** –ò—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å —Ç–æ–ª—å–∫–æ —Ñ–ª–∞–≥ –±–µ–∑ –∑–Ω–∞—á–µ–Ω–∏—è: `["-resetUserDefaults"]`
- **–ö–æ–¥ –æ–±—Ä–∞–±–æ—Ç–∫–∏ –≤ AppDelegate:**
  ```swift

### 2026-02-05: –ö–ª–∞–≤–∏–∞—Ç—É—Ä–Ω—ã–µ –∫–Ω–æ–ø–∫–∏ –¥–æ–ª–∂–Ω—ã –ø–æ—è–≤–ª—è—Ç—å—Å—è —Å–∏–Ω—Ö—Ä–æ–Ω–Ω–æ —Å –∫–ª–∞–≤–∏–∞—Ç—É—Ä–æ–π
- **–ü—Ä–æ–±–ª–µ–º–∞:** –ö–Ω–æ–ø–∫–∏ –Ω–∞–¥ –∫–ª–∞–≤–∏–∞—Ç—É—Ä–æ–π ¬´–ø–∞–¥–∞—é—Ç —Å–≤–µ—Ä—Ö—É¬ª –ø—Ä–∏ –ø–µ—Ä–≤–æ–º –ø–æ–∫–∞–∑–µ –∏ –ø–æ—è–≤–ª—è—é—Ç—Å—è —Å –∑–∞–¥–µ—Ä–∂–∫–æ–π –ø—Ä–∏ –ø–æ—Å–ª–µ–¥—É—é—â–∏—Ö.
- **–ü—Ä–∏—á–∏–Ω–∞:** –í –æ–±—Ä–∞–±–æ—Ç—á–∏–∫–µ `UIResponder.keyboardWill*` –∑–∞–º–µ–Ω–∏–ª–∏ —Å–∏–Ω—Ö—Ä–æ–Ω–Ω—ã–π –≤—ã–∑–æ–≤ –Ω–∞ `Task { @MainActor }`, —á—Ç–æ –¥–æ–±–∞–≤–∏–ª–æ –∑–∞–¥–µ—Ä–∂–∫—É –Ω–∞ –æ–¥–∏–Ω runloop‚Äë—Ç–∏–∫ –∏ —Ä–∞—Å—Å–∏–Ω—Ö—Ä–æ–Ω–∏–ª–æ –∞–Ω–∏–º–∞—Ü–∏–∏.
- **–†–µ—à–µ–Ω–∏–µ:** –û–±—Ä–∞–±–∞—Ç—ã–≤–∞—Ç—å –Ω–æ—Ç–∏—Ñ–∏–∫–∞—Ü–∏–∏ —Å–∏–Ω—Ö—Ä–æ–Ω–Ω–æ –Ω–∞ `.main` (–∏—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å `MainActor.assumeIsolated` –≤–Ω—É—Ç—Ä–∏ –∑–∞–º—ã–∫–∞–Ω–∏—è), –Ω–µ –ø–µ—Ä–µ–Ω–æ—Å–∏—Ç—å –æ–±—Ä–∞–±–æ—Ç–∫—É –≤ –æ—Ç–¥–µ–ª—å–Ω—ã–π `Task`, –µ—Å–ª–∏ –º—ã —É–∂–µ –Ω–∞ –≥–ª–∞–≤–Ω–æ–π –æ—á–µ—Ä–µ–¥–∏.
- **–£—Ä–æ–∫:** –î–ª—è —Å–∏–Ω—Ö—Ä–æ–Ω–∏–∑–∞—Ü–∏–∏ –∞–Ω–∏–º–∞—Ü–∏–π —Å –∫–ª–∞–≤–∏–∞—Ç—É—Ä–æ–π –Ω–µ –¥–æ–±–∞–≤–ª—è—Ç—å –∞—Å–∏–Ω—Ö—Ä–æ–Ω–Ω—ã–µ hop'—ã –≤ –æ–±—Ä–∞–±–æ—Ç—á–∏–∫–∏ `keyboardWillShow/Change/Hide`.
  if ProcessInfo.processInfo.arguments.contains("-resetUserDefaults") {
      UserDefaults.standard.removePersistentDomain(forName: bundleID)
  }
  ```
- **–£—Ä–æ–∫:** Launch arguments - —ç—Ç–æ –º–∞—Å—Å–∏–≤ —Å—Ç—Ä–æ–∫, –∫–∞–∂–¥—ã–π —Ñ–ª–∞–≥ - –æ—Ç–¥–µ–ª—å–Ω—ã–π —ç–ª–µ–º–µ–Ω—Ç

### 2026-01-24: UI —Ç–µ—Å—Ç—ã - Disabled TextField –Ω–µ–ª—å–∑—è —Ä–µ–¥–∞–∫—Ç–∏—Ä–æ–≤–∞—Ç—å
- **–û—à–∏–±–∫–∞:** `Failed to synthesize event: Neither element nor any descendant has keyboard focus`
- **–ü—Ä–∏—á–∏–Ω–∞:** –ü–æ —É–º–æ–ª—á–∞–Ω–∏—é `selectedRow = .total`, –ø–æ—ç—Ç–æ–º—É –≤—Å–µ `row_X_bet_size_text_field` –∑–∞–±–ª–æ–∫–∏—Ä–æ–≤–∞–Ω—ã (disabled)
- **–†–µ—à–µ–Ω–∏–µ:** –ò—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å –ø–æ–ª—è, –∫–æ—Ç–æ—Ä—ã–µ –≤—Å–µ–≥–¥–∞ enabled:
  - `total_row_bet_size_text_field` - –≤—Å–µ–≥–¥–∞ enabled –∫–æ–≥–¥–∞ selectedRow == .total
  - `row_X_coefficient_text_field` - –≤—Å–µ–≥–¥–∞ enabled
- **–£—Ä–æ–∫:** –ü–µ—Ä–µ–¥ –Ω–∞–ø–∏—Å–∞–Ω–∏–µ–º UI —Ç–µ—Å—Ç–æ–≤ –∏–∑—É—á–∏—Ç—å –ª–æ–≥–∏–∫—É –ø—Ä–∏–ª–æ–∂–µ–Ω–∏—è - –∫–∞–∫–∏–µ –ø–æ–ª—è enabled/disabled –≤ –∫–∞–∫–æ–º —Å–æ—Å—Ç–æ—è–Ω–∏–∏

### 2026-01-24: UI —Ç–µ—Å—Ç—ã - –ú–Ω–æ–∂–µ—Å—Ç–≤–µ–Ω–Ω—ã–µ —ç–ª–µ–º–µ–Ω—Ç—ã —Å –æ–¥–∏–Ω–∞–∫–æ–≤—ã–º —Ç–µ–∫—Å—Ç–æ–º
- **–û—à–∏–±–∫–∞:** `Multiple matching elements found` –¥–ª—è –∫–Ω–æ–ø–∫–∏ "Close"
- **–ü—Ä–∏—á–∏–Ω–∞:** –ù–∞ –ø–æ—Å–ª–µ–¥–Ω–µ–π —Å—Ç—Ä–∞–Ω–∏—Ü–µ –æ–Ω–±–æ—Ä–¥–∏–Ω–≥–∞ –¥–≤–µ –∫–Ω–æ–ø–∫–∏: X (Close onboarding) –∏ –æ—Å–Ω–æ–≤–Ω–∞—è (Close)
- **–†–µ—à–µ–Ω–∏–µ:** 
  - –î–æ–±–∞–≤–∏—Ç—å —É–Ω–∏–∫–∞–ª—å–Ω—ã–π `accessibilityLabel` –¥–ª—è –∫–Ω–æ–ø–∫–∏ X: `"Close onboarding"`
  - –ò—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å NSPredicate –¥–ª—è –ø–æ–∏—Å–∫–∞: `app.buttons.matching(NSPredicate(format: "label == 'Close'"))`
- **–£—Ä–æ–∫:** –í—Å–µ–≥–¥–∞ –¥–æ–±–∞–≤–ª—è—Ç—å —É–Ω–∏–∫–∞–ª—å–Ω—ã–µ accessibilityLabel –¥–ª—è –∫–Ω–æ–ø–æ–∫ —Å –∏–∫–æ–Ω–∫–∞–º–∏

### 2026-01-24: UI —Ç–µ—Å—Ç—ã - Image —Å onTapGesture
- **–û—à–∏–±–∫–∞:** –ö–Ω–æ–ø–∫–∏ +/- (Image —Å onTapGesture) –Ω–µ –Ω–∞—Ö–æ–¥—è—Ç—Å—è –∫–∞–∫ buttons
- **–ü—Ä–∏—á–∏–Ω–∞:** Image —Å onTapGesture - —ç—Ç–æ –Ω–µ Button, –∞ Image –≤ accessibility tree
- **–†–µ—à–µ–Ω–∏–µ:** –ò—Å–∫–∞—Ç—å —á–µ—Ä–µ–∑ `app.images["identifier"]` –≤–º–µ—Å—Ç–æ `app.buttons["identifier"]`
- **–£—Ä–æ–∫:** Image —Å onTapGesture –∏ accessibilityIdentifier –∏—â–µ—Ç—Å—è —á–µ—Ä–µ–∑ `app.images`, –Ω–µ `app.buttons`

### 2026-01-24: XCTestCase setUp/tearDown –∏ MainActor –≤ Swift 6
- **–û—à–∏–±–∫–∞:** `Main actor-isolated property 'app' can not be mutated from a nonisolated context` –≤ `setUpWithError()`
- **–ü—Ä–∏—á–∏–Ω–∞:** –ú–µ—Ç–æ–¥—ã `setUpWithError()` –∏ `tearDownWithError()` –Ω–µ –Ω–∞—Å–ª–µ–¥—É—é—Ç `@MainActor` –∏–∑–æ–ª—è—Ü–∏—é –æ—Ç –∫–ª–∞—Å—Å–∞
- **–†–µ—à–µ–Ω–∏–µ:** –ò—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å async –≤–µ—Ä—Å–∏–∏ –º–µ—Ç–æ–¥–æ–≤:
  ```swift
  @MainActor
  final class MyUITests: XCTestCase {
      override func setUp() async throws {
          try await super.setUp()
          app = XCUIApplication()
      }
      
      override func tearDown() async throws {
          app = nil
          try await super.tearDown()
      }
  }
  ```
- **–£—Ä–æ–∫:** –í Swift 6 –¥–ª—è `@MainActor` —Ç–µ—Å—Ç–æ–≤—ã—Ö –∫–ª–∞—Å—Å–æ–≤ –∏—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å `setUp() async throws` –≤–º–µ—Å—Ç–æ `setUpWithError() throws`

### 2026-01-24: Swift Testing - Race Condition –≤ —Ç–µ—Å—Ç–∞—Ö —Å UserDefaults
- **–û—à–∏–±–∫–∞:** –¢–µ—Å—Ç—ã –ø–∞–¥–∞—é—Ç —Å–ª—É—á–∞–π–Ω—ã–º –æ–±—Ä–∞–∑–æ–º –∏–∑-–∑–∞ race condition –ø—Ä–∏ —Ä–∞–±–æ—Ç–µ —Å UserDefaults
- **–ü—Ä–∏—á–∏–Ω–∞:** Swift Testing –ø–æ —É–º–æ–ª—á–∞–Ω–∏—é –≤—ã–ø–æ–ª–Ω—è–µ—Ç —Ç–µ—Å—Ç—ã –ø–∞—Ä–∞–ª–ª–µ–ª—å–Ω–æ, –∞ UserDefaults - shared state
- **–†–µ—à–µ–Ω–∏–µ:** –î–æ–±–∞–≤–∏—Ç—å `@Suite(.serialized)` –¥–ª—è –ø–æ—Å–ª–µ–¥–æ–≤–∞—Ç–µ–ª—å–Ω–æ–≥–æ –≤—ã–ø–æ–ª–Ω–µ–Ω–∏—è —Ç–µ—Å—Ç–æ–≤:
  ```swift
  @MainActor
  @Suite(.serialized)
  struct RootViewModelTests { ... }
  ```
- **–£—Ä–æ–∫:** –¢–µ—Å—Ç—ã, —Ä–∞–±–æ—Ç–∞—é—â–∏–µ —Å shared state (UserDefaults, Keychain, —Ñ–∞–π–ª—ã), –¥–æ–ª–∂–Ω—ã –≤—ã–ø–æ–ª–Ω—è—Ç—å—Å—è –ø–æ—Å–ª–µ–¥–æ–≤–∞—Ç–µ–ª—å–Ω–æ —á–µ—Ä–µ–∑ `@Suite(.serialized)`

### 2026-01-24: SwiftUI - Invalid frame dimension –≤ keyboard toolbar (–ë–ê–ì SWIFTUI)
- **–û—à–∏–±–∫–∞:** `Invalid frame dimension (negative or non-finite)` runtime warning –≤ UI —Ç–µ—Å—Ç–∞—Ö –ø—Ä–∏ —Ä–∞–±–æ—Ç–µ —Å –∫–ª–∞–≤–∏–∞—Ç—É—Ä–æ–π
- **–ü—Ä–∏—á–∏–Ω–∞:** **–ò–∑–≤–µ—Å—Ç–Ω—ã–π –±–∞–≥ SwiftUI** - `ToolbarItemGroup(placement: .keyboard)` –≤—ã–∑—ã–≤–∞–µ—Ç runtime warning –ø—Ä–∏ –ø–æ—è–≤–ª–µ–Ω–∏–∏/–∏—Å—á–µ–∑–Ω–æ–≤–µ–Ω–∏–∏ –∫–ª–∞–≤–∏–∞—Ç—É—Ä—ã –Ω–µ–∑–∞–≤–∏—Å–∏–º–æ –æ—Ç —Å–æ–¥–µ—Ä–∂–∏–º–æ–≥–æ
- **–ü–æ–ø—ã—Ç–∫–∏ –∏—Å–ø—Ä–∞–≤–ª–µ–Ω–∏—è (–ù–ï –†–ê–ë–û–¢–ê–Æ–¢):**
  - `Spacer(minLength: 0)` –≤–º–µ—Å—Ç–æ `Spacer()` - –Ω–µ –ø–æ–º–æ–≥–∞–µ—Ç
  - `.fixedSize()` –Ω–∞ –∫–Ω–æ–ø–∫–∞—Ö - –Ω–µ –ø–æ–º–æ–≥–∞–µ—Ç
  - –£–±—Ä–∞—Ç—å –∞–Ω–∏–º–∞—Ü–∏–∏ - –Ω–µ –ø–æ–º–æ–≥–∞–µ—Ç
  - –ò—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å –æ—Ç–¥–µ–ª—å–Ω—ã–µ `ToolbarItem` –≤–º–µ—Å—Ç–æ `ToolbarItemGroup` - –Ω–µ –ø–æ–º–æ–≥–∞–µ—Ç
  - `.frame(minWidth: 0)` –Ω–∞ –∫–æ–º–ø–æ–Ω–µ–Ω—Ç–∞—Ö - –Ω–µ –ø–æ–º–æ–≥–∞–µ—Ç
- **–ï–¥–∏–Ω—Å—Ç–≤–µ–Ω–Ω–æ–µ —Ä–µ—à–µ–Ω–∏–µ:** –£–±—Ä–∞—Ç—å keyboard toolbar –ø–æ–ª–Ω–æ—Å—Ç—å—é - —Ç–æ–≥–¥–∞ warnings –∏—Å—á–µ–∑–∞—é—Ç
- **–†–µ–∫–æ–º–µ–Ω–¥–∞—Ü–∏—è:** **–ò–≥–Ω–æ—Ä–∏—Ä–æ–≤–∞—Ç—å warning** - —ç—Ç–æ runtime diagnostic –æ—Ç —Å–∏–º—É–ª—è—Ç–æ—Ä–∞, –Ω–µ –≤–ª–∏—è–µ—Ç –Ω–∞ —Ä–∞–±–æ—Ç—É –ø—Ä–∏–ª–æ–∂–µ–Ω–∏—è, —Ç–µ—Å—Ç—ã –ø—Ä–æ—Ö–æ–¥—è—Ç
- **–°—Å—ã–ª–∫–∞:** https://developer.apple.com/forums/thread/709656
- **–£—Ä–æ–∫:** `ToolbarItemGroup(placement: .keyboard)` –≤ SwiftUI –∏–º–µ–µ—Ç –∏–∑–≤–µ—Å—Ç–Ω—ã–π –±–∞–≥ —Å "Invalid frame dimension" warnings. –≠—Ç–æ –Ω–µ –æ—à–∏–±–∫–∞ –ø—Ä–∏–ª–æ–∂–µ–Ω–∏—è

---

**–ü–æ—Å–ª–µ–¥–Ω–µ–µ –æ–±–Ω–æ–≤–ª–µ–Ω–∏–µ:** 2026-01-31

### 2026-02-01: Destination ID –∏ —Å–±–æ–∏ CoreSimulator
- **–ù–∞–±–ª—é–¥–µ–Ω–∏–µ:** Destination ID –∏–∑ AGENTS.md –º–æ–∂–µ—Ç –±—ã—Ç—å machine-specific; –≤ –∞–≥–µ–Ω—Ç—Å–∫–∏—Ö –æ–∫—Ä—É–∂–µ–Ω–∏—è—Ö –≤–æ–∑–º–æ–∂–Ω—ã –Ω–µ—Å–æ–æ—Ç–≤–µ—Ç—Å—Ç–≤–∏—è.
- **–°–∏–º–ø—Ç–æ–º:** `xcodebuild test` –ø–∞–¥–∞–µ—Ç —Å –æ—à–∏–±–∫–∞–º–∏ CoreSimulator (`simdiskimaged` –Ω–µ –æ—Ç–≤–µ—á–∞–µ—Ç / runtimes –Ω–µ –æ–±–Ω–∞—Ä—É–∂–µ–Ω—ã) –¥–∞–∂–µ –ø—Ä–∏ –Ω–∞–ª–∏—á–∏–∏ —É—Å—Ç—Ä–æ–π—Å—Ç–≤–∞ –≤ `simctl list devices available`.
- **–ü—Ä–æ–≤–µ—Ä–∫–∞:** –í—Å–µ–≥–¥–∞ –∑–∞–ø—É—Å–∫–∞—Ç—å `xcrun simctl list devices available` –∏ –ø–æ–¥—Ç–≤–µ—Ä–∂–¥–∞—Ç—å –Ω–∞–ª–∏—á–∏–µ —É–∫–∞–∑–∞–Ω–Ω–æ–≥–æ ID –¥–æ fallback.
- **Fallback:** –ò—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å –¥—Ä—É–≥–æ–π iPhone Simulator —Ç–æ–ª—å–∫–æ –µ—Å–ª–∏ ID –∏–∑ AGENTS.md –Ω–µ–¥–æ—Å—Ç—É–ø–µ–Ω; AGENTS.md –Ω–µ –º–µ–Ω—è—Ç—å –∞–≤—Ç–æ–º–∞—Ç–∏—á–µ—Å–∫–∏.
- **CI/–∞–≥–µ–Ω—Ç—ã:** –û–∫—Ä—É–∂–µ–Ω–∏—è –±–µ–∑ –∫–æ—Ä—Ä–µ–∫—Ç–Ω–æ –∑–∞–ø—É—â–µ–Ω–Ω–æ–≥–æ CoreSimulator –º–æ–≥—É—Ç –¥–∞–≤–∞—Ç—å –ª–æ–∂–Ω—ã–µ –ø–∞–¥–µ–Ω–∏—è —Ç–µ—Å—Ç–æ–≤ –±–µ–∑ –ø—Ä–æ–±–ª–µ–º –≤ –∫–æ–¥–µ.
